test/listen.4.test.js

```diff
@@ -22,63 +22,70 @@ before(async function () {
   [, localhostForURL] = await helper.getLoopbackHost()
 })
 
-test('listen twice on the same port without callback rejects', t => {
+test('listen twice on the same port without callback rejects', (t, done) => {
   t.plan(1)
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
+  t.after(() => fastify.close())
 
   fastify.listen({ port: 0 })
     .then(() => {
-      const s2 = Fastify()
-      t.teardown(s2.close.bind(s2))
-      s2.listen({ port: fastify.server.address().port })
+      const server2 = Fastify()
+      t.after(() => server2.close())
+      server2.listen({ port: fastify.server.address().port })
         .catch(err => {
-          t.ok(err)
+          t.assert.ok(err)
+          done()
         })
     })
-    .catch(err => t.error(err))
+    .catch(err => {
+      t.assert.ifError(err)
+    })
 })
 
-test('listen twice on the same port without callback rejects with (address)', t => {
+test('listen twice on the same port without callback rejects with (address)', (t, done) => {
   t.plan(2)
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
+  t.after(() => fastify.close())
   fastify.listen({ port: 0 })
     .then(address => {
-      const s2 = Fastify()
-      t.teardown(s2.close.bind(s2))
-      t.equal(address, `http://${localhostForURL}:${fastify.server.address().port}`)
-      s2.listen({ port: fastify.server.address().port })
+      const server2 = Fastify()
+      t.after(() => server2.close())
+      t.assert.strictEqual(address, `http://${localhostForURL}:${fastify.server.address().port}`)
+
+      server2.listen({ port: fastify.server.address().port })
         .catch(err => {
-          t.ok(err)
+          t.assert.ok(err)
+          done()
         })
     })
-    .catch(err => t.error(err))
+    .catch(err => {
+      t.assert.ifError(err)
+    })
 })
 
 test('listen on invalid port without callback rejects', t => {
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
+  t.after(() => fastify.close())
   return fastify.listen({ port: -1 })
     .catch(err => {
-      t.ok(err)
+      t.assert.ok(err)
       return true
     })
 })
 
-test('listen logs the port as info', t => {
+test('listen logs the port as info', async t => {
   t.plan(1)
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
+  t.after(() => fastify.close())
 
   const msgs = []
   fastify.log.info = function (msg) {
     msgs.push(msg)
   }
 
-  fastify.listen({ port: 0 })
+  await fastify.listen({ port: 0 })
```

Comment:
we can remove the `then` here

Reply:
I have updated the test, see commit here: https://github.com/fastify/fastify/commit/fae9eb86c02b47380b4448e4005c7ed5500fc2b5